src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -40,8 +40,11 @@
 SEARCH_PATHS = ("/usr/bin", "/bin", "/usr/local/bin", "/opt/homebrew/bin")
 
 
+Tbp = TypeVar("Tbp", bound="BinaryPath")
```

Comment:
Give me exactly what you want and I'll do it.

Just using Python core ~standards here: https://github.com/python/typeshed/blob/main/stdlib/typing.pyi#L577

So `T`, but since TypeVar are global, scoped to the intended use exactly here, `Tbp` -> TypeBinaryPath.

Reply:
Better - give me the rule I can add to the style guide (https://www.pantsbuild.org/docs/style-guide#type-hints) to avoid this sort of noise in the future.

Reply:
`BinaryPathType`?

Reply:
Yeah - my point is this is all totally arbitrary; so I'd like @thejcannon to call his shot and name the algorithm and I'll fix this and update the style guide. When reviewers are worried about stuff like this, it distracts from where real review power should go.

Reply:
Prior examples in the codebase are of the form `BoundClass``T` (e.g. `AddressMapT` in `internals/mapper.py`).

Reply:
(and I'd be happy to adopt the `BoundClassT` rule into the style guide, since we're pretty consistent about it elsewhere in the codebase)

Reply:
Here are the facts:
```
$ git grep "TypeVar(" | sed -r -e 's|.*\sTypeVar\("([^"]+)".*\).*|\1|' | sort | uniq -c | sort -n
      1 AddressMapT
      1 _BoolDefault
      1 C
      1 _CD
      1 DefaultT
      1 _DefaultT
      1 _E
      1 _EHPB
      1 _EnumT
      1 ErrorT
      1 _ExternalHelmPlugin
      1 _FloatDefault
      1 FS
      1 _G
      1 _GoalT
      1 _GS
      1 _I
      1 _IntDefault
      1 _LintRequestT
      1 _ListMemberT
      1 _LockfileMetadataSubclass
      1 _O
      1 _OptT
      1 _Out0
      1 _Out1
      1 _Out2
      1 _Out3
      1 _Out4
      1 _Out5
      1 _Out6
      1 _Out7
      1 _Out8
      1 _Out9
      1 PartitionElementT
      1 PartitionMetadataT
      1 _R
      1 ReturnType
      1 _SS
      1 _StrDefault
      1 _SubsystemT
      1 _TAbstractOrderedSet
      1 _TargetGenerator
      1 Tbp
      1 T_co
      1 _TestFieldSetT
      1 ToolBaseT
      1 _V
      1 _ValueT
      2 _CoreRequestType
      2 F
      2 _F
      2 _FieldSetT
      2 _FilePartitioner
      2 _Input
      2 K
      2 _Output
      2 _TargetPartitioner
      2 V
      3 R
      7 _FS
     10 T
     12 _T
```

Reply:
@jsirois I don't get to speak for the entire codebase, so I reject your instruction. What I can do is say that, to me, a reviewer on your PR and maintainer of the code. `Tbp` is confusing to read. Pointing out distracting names is not a distraction. Code is read much more than it is written or executed, so I firmly believe discussions on readability of code are important.

That being said, let's make this more readable (perhaps as @chrisjrn suggests `BinaryPathT`) and we can discuss the style guide entry in a PR that updates it.

Reply:
(I'm working on the style guide entry now)

Reply:
> `Tbp` means nothing to me. Can we spell this out a bit more?

`Tbp` means nothing to _you_ and yet you asked we to make it mean more to you. Now we is me, and clearly I understood `Tbp`; so, as I pointed out - this is arbitrary stuff. Its eye of the beholder. In those cases its much more useful as a reviewer to say what you mean and spell out an example you find more meaningful instead of leaving me to guess. Further, I still contend this stuff is not useful to waste time reviewing and I think you actually definitely agree; otherwise you wouldn't want black, etc; so codifying the mechanical stuff in code style is the way to go - save brain power for finding bugs, etc. Chris stepped in for the latter. Thank you for confirming his suggestion on the former.

Reply:
Fixed.